hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -223,11 +227,21 @@ public TransferList netTransfersInTxn() {
 			var token = tokensTouched[i];
 			if (i == 0 || !token.equals(tokensTouched[i - 1])) {
 				var netTransfersHere = netTokenTransfers.get(token);
+				var uniqueTransfersHere = uniqueTokenTransfers.get(token);
 				purgeZeroAdjustments(netTransfersHere);
-				all.add(TokenTransferList.newBuilder()
-						.setToken(token)
-						.addAllTransfers(netTransfersHere.getAccountAmountsList())
-						.build());
+
+				if (!isFungibleToken(token) && uniqueTransfersHere != null) {
+					all.add(TokenTransferList.newBuilder()
+							.setToken(token)
+							.addAllTransfers(netTransfersHere.getAccountAmountsList())
```

Comment:
For NFT transfers we will not populate `account amount list` we must populate only `nftTransferList`

Reply:
Confirmed and corrected!